src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleLogFormatter.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class JsonConsoleLogFormatter : ConsoleFormatter, IDisposable
+    {
+        private IDisposable _optionsReloadToken;
+
+        public JsonConsoleLogFormatter(IOptionsMonitor<JsonConsoleFormatterOptions> options)
+            : base (ConsoleFormatterNames.Json)
+        {
+            FormatterOptions = options.CurrentValue;
+            ReloadLoggerOptions(options.CurrentValue);
+            _optionsReloadToken = options.OnChange(ReloadLoggerOptions);
+        }
+
+        public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeProvider scopeProvider, TextWriter textWriter)
+        {
+            LogLevel logLevel = logEntry.LogLevel;
+            string category = logEntry.Category;
+            int eventId = logEntry.EventId.Id;
+            string message = logEntry.Formatter(logEntry.State, logEntry.Exception);
+            Exception exception = logEntry.Exception;
+            const int DefaultBufferSize = 1024;
+            var output = new ArrayBufferWriter<byte>(DefaultBufferSize);
```

Comment:
`ArrayBufferWriter` always creates new arrays, and resizes them (i.e. allocates more memory) as needed. I wonder if we should reuse [https://github.com/dotnet/runtime/blob/master/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/PooledByteBufferWriter.cs](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/PooledByteBufferWriter.cs) here instead, so the memory is pooled.

Reply:
Sure but PooledByteBufferWriter is internal to System.Text.Json

Reply:
https://github.com/dotnet/runtime/issues/33598#issuecomment-599144041